python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -213,14 +213,33 @@ def dist_inference_one_layer(layer_id, g, dataloader, target_ntypes, layer, get_
     -------
         dict of Tensors : the inferenced tensors.
     """
```

Comment:
The implementation of dist_inference_one_layer is very tricky. You need to take a look at dist_inference @L311, when i > 0, the get_input_embeds is another function, it is not accessing input features.

Thus, we can check the layer_id to decide whether we need to do the dataloader

Reply:
I think its alright. It true that the fix is unnecessary when layer_id >0 and we can add an if-else to check but that will make the code messy.

Reply:
Fixed